test/common/router/scoped_rds_test.cc

```diff
@@ -203,48 +165,10 @@ route_configuration_name: foo_routes
   Protobuf::RepeatedPtrField<ProtobufWkt::Any> resources;
   parseScopedRouteConfigurationFromYaml(*resources.Add(), config_yaml);
   parseScopedRouteConfigurationFromYaml(*resources.Add(), config_yaml);
-  EXPECT_THROW_WITH_MESSAGE(subscription().onConfigUpdate(resources, "1"), EnvoyException,
+  EXPECT_THROW_WITH_MESSAGE(subscription_callbacks_->onConfigUpdate(resources, "1"), EnvoyException,
                             "duplicate scoped route configuration foo_scope found");
 }
 
-// Tests that defining an invalid cluster in the SRDS config results in an error.
-TEST_F(ScopedRdsTest, UnknownCluster) {
```

Comment:
Hmm, this is no longer the case?

Reply:
This validation is done by the subscription factory and tested at https://github.com/envoyproxy/envoy/blob/master/test/common/config/subscription_factory_test.cc#L313. It's not really possible to test this with a mock subscription factory as a result.